lib/pxutil.js

```diff
@@ -269,20 +269,34 @@ function getTokenObject(cookie, delimiter = ':') {
 
 function getGraphqlData(req) {
     const isGraphqlPath = req.path.includes('graphql') || req.originalUrl.includes('graphql');
-    if (!isGraphqlPath) {
+    if (!isGraphqlPath || !req.body) {
         return null;
     }
 
-    const query = req.body ? req.body['query'] : null;
-    if (!query) {
-        return new GraphqlData(GraphqlOperationType.QUERY);
+    const { body, query } = getGraphqlBodyAndQuery(req);
+    if (!body || !query) {
+        return null;
     }
 
     const operationType = extractGraphqlOperationType(query);
-    const operationName = extractGraphqlOperationName(query, req.body['operationName'])
+    const operationName = extractGraphqlOperationName(query, body['operationName'])
     return new GraphqlData(operationType, operationName);
 }
 
+function getGraphqlBodyAndQuery(req) {
```

Comment:
- Don't we need to clone the request to allow multiple reading?
- What about supporting bodySizeLimit?